### ISSUE TITLE
RP listener throws error while run tests

### ISSUE THREAD

**Issue by KursLabIgor** [reportportal/reportportal#295]
We have amount 160 test based in nunit3 , and while run we got error like 
14:06:09 ReportPortal exception was thrown.
14:06:09 System.NullReferenceException: Object reference not set to an instance of an object.
14:06:09    at ReportPortal.NUnitExtension.ReportPortalListener.FinishTest(XmlDocument xmlDoc)

Do we have some option how to get more details what cause error?

**Comment by nvborisenko** [comment]
Could you please provide the nunit version and versions of nuget packages for Report Portal? If it possible it would be great to see `results.xml` file.

**Comment by KursLabIgor** [comment]
@nvborisenko 
ReportPortal.Client v2.0.0-beta3
ReportPortal.NUnit v 3.3.0-beta1
ReportPortal.Shared v2.0.0-beta1
About result.xml not possible to provide you. it was erased.

**Comment by nvborisenko** [comment]
Is this issue always reproducible? Or from time to time? If yes - I guess issue is related to network connection.
I see you use retry functionality. Unfortunately I did test this functionality with RP integration. Are you able to execute your tests with turned off Retry attribute. Based on exception I can say that nunit generates xml file with results in some different format. If it will work then I will see deeper into the issue.

There are no internal logs yet :(

**Comment by KursLabIgor** [comment]
@nvborisenko until now we have reproduce issue 2 time. i'll try catch result.xml file during next run. 
Remove retry is painful for us, so bottom line trying get more details

**Comment by nvborisenko** [comment]
I hope this is related to broken reporting (ReportPortal.Shared library). As I know you use beta version, you know what is beta. Thank you for the issue reporting.

Let's leave this issue to be opened.

**Comment by KursLabIgor** [comment]
@nvborisenko do you have some ETA about release of agent?

**Comment by nvborisenko** [comment]
It's not clear what is a root of this issue. https://github.com/reportportal/agent-net-nunit/blob/master/ReportPortal.NUnitExtension/ReportPortalListener.Test.cs here you can see possible NullReferenceException, but we need understand where exactly it's happening. Based on code I see that `id`, 'parentId', 'name' or 'fullname' might be null. But I cannot understand why it might be null. To clarify it we need understand what features of nunit you use and try to reproduce this issue.

**Comment by KursLabIgor** [comment]
@nvborisenko we have two problems here: 
1. Issue reproduce without stable scenario.
2. We have complicated framework, but i'll consult with team and think about it how we can help here. 
But i think firstly need to find at least stable scenario. any way i'll back with updates asap.

**Comment by KursLabIgor** [comment]
@nvborisenko we have detected such behavior when added nunit test attribute Timeout.
Examaple
[Category("SendTo"), Category("UWP"), Order(1), Timeout(600 * 1000)]
when test canceled by time out we got error like
'#1: Test cancelled by user - Timeout!'
ReportPortal exception was thrown.
System.NullReferenceException: Object reference not set to an instance of an object.
   at ReportPortal.NUnitExtension.ReportPortalListener.FinishTest(XmlDocument xmlDoc)
